### PR TITLE
[refactor] Removing Joomla code, more table checks

### PR DESCRIPTION
**Pull request by zooley**
Removing Joomla-based code and adding more table existence checks. Also
having methods return boolean values in accordance to their docblock.

### CODE REVIEW

File: src/Content/Migration/Base.php [hunk 1]
```diff
@@ -192,10 +192,10 @@ public function getMWDBO()
 	 **/
 	private function getRootCredentials()
 	{
-		$secrets   = DS . 'etc'  . DS . 'hubzero.secrets';
-		$conf_file = DS . 'root' . DS . '.my.cnf';
-		$hub_maint = DS . 'etc'  . DS . 'mysql' . DS . 'hubmaint.cnf';
-		$deb_maint = DS . 'etc'  . DS . 'mysql' . DS . 'debian.cnf';
+		$secrets   = DIRECTORY_SEPARATOR . 'etc'  . DIRECTORY_SEPARATOR . 'hubzero.secrets';
```
Comment (zweidner) [inline src/Content/Migration/Base.php]: Do we still need directory separator constants, or is now not quite the time to remove them all?

File: src/Content/Migration/Base.php [hunk 2]
```diff
@@ -409,8 +409,10 @@ public function addComponentEntry($name, $option=null, $enabled=1, $params='', $
 			$query .= " VALUES ('{$name}', 'option={$option}', 0, 0, 'option={$option}', '{$name}', '{$option}', {$ordering}, '', 0, ".$this->baseDb->quote($params).", {$enabled})";
 			$this->baseDb->setQuery($query);
 			$this->baseDb->query();
+
+			return true;
 		}
-		else
+		elseif ($this->baseDb->tableExists('#__extensions'))
```
Comment (zweidner) [inline src/Content/Migration/Base.php]: > Note: Note that elseif and else if will only be considered exactly the same when using curly brackets as in the above example. When using a colon to define your if/elseif conditions, you must not separate else if into two words, or PHP will fail with a parse error.

Stylistically, should we stick to combining to elseif in this case?

File: src/Content/Migration/Base.php (hunk 1 above)
Comment (zooley) [reply src/Content/Migration/Base.php]: We don't really need them, I guess. I was just trying to get rid of usage of CMS-specific constants.

File: src/Content/Migration/Base.php (hunk 2 above)
Comment (zooley) [reply src/Content/Migration/Base.php]: Yes, we should settle on the format that works best in both cases.

**Pull request merged**